Explorer/Assets/DCL/UI/ConnectionStatusPanel/Badge/BadgeView.cs

```diff
@@ -0,0 +1,32 @@
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace DCL.UI.ConnectionStatusPanel.Badge
+{
+    public class BadgeView : MonoBehaviour
+    {
+        [SerializeField] private RectTransform self = null!;
+        [SerializeField] private TMP_Text tmpText = null!;
+        [SerializeField] private List<Image> backgroundPieces = new ();
+
+        public void UpdateText(string text, float? width = null)
+        {
+            tmpText.SetText(text);
+
+            if (width is { } w && Mathf.Approximately(w, 0) == false)
```

Comment:
Not clear what this line doing, can you refactor? `if (width != null && Mathf....)`

Reply:
same, it says "if width is object (not null) then it's 'w' and check if the value is not equal to zero"

Reply:
Yeah, but this kind of syntax makes sense when you can validate a property, like: `if (obj is { Enabled })`, but expressing just `{ }` pollutes the reading. Its more clear if declared as: `if (width != null)`